src/QirTools/pyqir/qirlib/src/emit/qir/array1d.rs

```diff
@@ -0,0 +1,237 @@
+        context,
+        index,
+        sub_result.as_basic_value_enum(),
+        element_raw_ptr_name.as_str(),
+    );
+
+    let element_result_ptr_name = format!("{}_result_{}", sub_result_name, index);
+    let target_type = context.types.array.ptr_type(AddressSpace::Generic);
+    let cast = context.builder.build_bitcast(
+        sub_result_element_ptr,
+        target_type,
+        element_result_ptr_name.as_str(),
+    );
+    cast
+}
+
+fn get_bitcast_qubit_pointer_element<'ctx>(
+    context: &Context<'ctx>,
+    i: u64,
+    sub_result: &BasicValueEnum<'ctx>,
+    sub_result_name: &str,
+) -> BasicValueEnum<'ctx> {
+    let element_raw_ptr_name = format!("{}_{}_raw", sub_result_name, i);
+    let sub_result_element_ptr = emit_array_get_element_ptr_1d(
+        context,
+        i,
+        sub_result.as_basic_value_enum(),
+        element_raw_ptr_name.as_str(),
+    );
+
+    let element_result_ptr_name = format!("{}_result_{}", sub_result_name, i);
+    let target_type = context.types.qubit;
+    let cast = context.builder.build_bitcast(
+        sub_result_element_ptr,
+        target_type.ptr_type(AddressSpace::Generic),
+        element_result_ptr_name.as_str(),
+    );
+    cast
+}
+
+fn get_bitcast_array_element<'ctx>(
+    context: &Context<'ctx>,
+    index: u64,
+    sub_result: &BasicValueEnum<'ctx>,
+    sub_result_name: &str,
+) -> BasicValueEnum<'ctx> {
+    let element_raw_ptr_name = format!("{}_{}_raw", sub_result_name, index);
+    let sub_result_element_ptr = emit_array_get_element_ptr_1d(
+        context,
+        index,
+        sub_result.as_basic_value_enum(),
+        element_raw_ptr_name.as_str(),
+    );
+
+    let element_result_ptr_name = format!("{}_result_{}", sub_result_name, index);
+    let target_type = context.types.array;
+    let cast = context.builder.build_bitcast(
+        sub_result_element_ptr,
+        target_type,
+        element_result_ptr_name.as_str(),
+    );
+    cast
+}
+
+pub fn get_bitcast_result_pointer_array_element<'ctx>(
+    context: &Context<'ctx>,
+    index: u64,
+    sub_result: &BasicValueEnum<'ctx>,
+    sub_result_name: &str,
+) -> BasicValueEnum<'ctx> {
+    let element_raw_ptr_name = format!("{}_{}_raw", sub_result_name, index);
+    let sub_result_element_ptr = emit_array_get_element_ptr_1d(
+        context,
+        index,
+        sub_result.as_basic_value_enum(),
+        element_raw_ptr_name.as_str(),
+    );
+
+    let element_result_ptr_name = format!("{}_result_{}", sub_result_name, index);
+    let target_type = context.types.result.ptr_type(AddressSpace::Generic);
+    let cast = context.builder.build_bitcast(
+        sub_result_element_ptr,
+        target_type,
+        element_result_ptr_name.as_str(),
+    );
+    cast
+}
+
+pub(crate) fn emit_empty_result_array_allocate1d<'ctx>(
+    context: &Context<'ctx>,
+    result_name: &str,
+) -> BasicValueEnum<'ctx> {
+    let results = emit_array_allocate1d(&context, 8, 0, &result_name[..]);
+    results
```

Comment:
```suggestion
    emit_array_allocate1d(&context, 8, 0, &result_name[..])
```

Reply:
Separated on purpose to give debugging/inspection attachment point